### PR TITLE
Formatter::TimeFormatter humanize

### PR DESCRIPTION
**Pull request by wayne-weibel**
## Summary

add formatter function for human readable time span - quality of life when generating reports against database

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/89710

## Testing done

- [x] *New code is covered by unit tests*
- local testing

## Screenshots
_Note: Optional_

## What areas of the site does it impact?

none

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by mjknight50 — DISMISSED** [review]
GTG

File: lib/formatters/time_formatter.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    # @param secs [Float|Integer] the number of seconds to format, eg: 940913.38729661
+    #
+    # @return [String] the human readable string
+    #
+    def self.humanize(secs)
+      [[60, :seconds], [60, :minutes], [24, :hours], [Float::INFINITY, :days]].map do |count, name|
+        if secs.positive?
+          secs, n = secs.divmod(count)
+
+          "#{n.to_i} #{name}" unless n.to_i.zero?
```
Comment (rmtolmach) [inline lib/formatters/time_formatter.rb:17]: Is there a chance that `secs` will ever equal `0`? if so, what will happen? Or is that covered in this line?

Comment (wayne-weibel) [reply lib/formatters/time_formatter.rb:17]: There is the chance, if it is a full minute, or 0 is passed - the result, for that iteration, is an empty string:
```
3.3.3 :015 > def humanize(secs)
3.3.3 :016 >   [[60, :seconds], [60, :minutes], [24, :hours], [Float::INFINITY, :days]].map do |count, name|
3.3.3 :017 >     if secs.positive?
3.3.3 :018 >       secs, n = secs.divmod(count)
3.3.3 :019 > 
3.3.3 :020 >       "#{n.to_i} #{name}" unless n.to_i.zero?
3.3.3 :021 >     end
3.3.3 :022 >   end.compact.reverse.join(' ')
3.3.3 :023 > end
 => :humanize 
3.3.3 :024 > 
3.3.3 :025 > humanize 0
 => "" 
3.3.3 :026 > humanize 60
 => "1 minutes" 
3.3.3 :027 > humanize 120
 => "2 minutes" 
3.3.3 :028 > humanize 3600
 => "1 hours" 
3.3.3 :029 > humanize 3660
 => "1 hours 1 minutes" 
3.3.3 :030 > humanize 3780
 => "1 hours 3 minutes" 
3.3.3 :031 >

```

Comment (rmtolmach) [reply lib/formatters/time_formatter.rb:17]: ok, and is an empty string acceptable?

**Comment by bosawt** [comment]
Does this PR differ from this? https://apidock.com/rails/ActionView/Helpers/DateHelper/distance_of_time_in_words

**Comment by wayne-weibel** [comment]
> Does this PR differ from this? https://apidock.com/rails/ActionView/Helpers/DateHelper/distance_of_time_in_words

This will break it down to an exact value, instead of using 'about' and 'less than'.  
And in our current Rails it seems this function is not available:
image

File: lib/formatters/time_formatter.rb (hunk 1 above)
Comment (wayne-weibel) [reply lib/formatters/time_formatter.rb]: For what I was using it for yes.  Assumption being that the caller would only do so if >0 and/or want to handle the empty string specific to situation.  I added a note to the docstring

**Review by mjknight50 — APPROVED** [review]
GTG

**Comment by bosawt** [comment]
> > Does this PR differ from this? https://apidock.com/rails/ActionView/Helpers/DateHelper/distance_of_time_in_words
> 
> This will break it down to an exact value, instead of using 'about' and 'less than'. And in our current Rails it seems this function is not available: image

We do have to access it, you have to include it in your class: `include ActionView::Helpers::DateHelper`.  Although, you're right, it's not exact. I assumed they'd have an 'exact' option or something but it looks like that's not the case

**Review by bosawt — APPROVED** [review]
looks good, I tested with various seconds combinations and it appears to work as expected

**Review by RachalCassity — APPROVED** [review]

**Pull request merged**